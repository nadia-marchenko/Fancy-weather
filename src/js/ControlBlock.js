import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import DropdownButton from 'react-bootstrap/DropdownButton';
import ButtonGroup from 'react-bootstrap/ButtonGroup';
import Dropdown from 'react-bootstrap/Dropdown';
import InputGroup from 'react-bootstrap/InputGroup';
import Form from 'react-bootstrap/Form';
import FormControl from 'react-bootstrap/FormControl';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Container from 'react-bootstrap/Container';



class ControlBlock extends Component {
  constructor() {
    super();
    this.state = {
      inputValue: ''
    };
  }

  updateInputValue(evt) {
    this.setState({
      inputValue: evt.target.value
    });
  }

  render() {
    let celsiumTypeDegree = 'celsium';
    let farengeitTypeDegree = 'farengeit';

    return (
    <>
      <div className="control-block">
        <Container>
          <Row>
            <Col lg={4} xs={12} md={6}>
              <div className="main-buttons">
                <Button variant="outline-secondary" onClick={() => { this.props.updateBackground()}}>Update</Button>{' '}
                <DropdownButton as={ButtonGroup} title="LAN" id="dropdown-variants-Secondary" variant="outline-secondary">
                  <Dropdown.Item eventKey="1">EN</Dropdown.Item>
                  <Dropdown.Item eventKey="2">RU</Dropdown.Item>
                  <Dropdown.Item eventKey="2">BE</Dropdown.Item>
                </DropdownButton>
                <Dropdown as={ButtonGroup} variant="outline-secondary" className="temperature-change-buttons">
                  <Button variant="outline-secondary" onClick={() => { this.props.toCelsium(celsiumTypeDegree)}}>°C</Button>
                  <Button variant="outline-secondary" onClick={() => { 
                    this.props.toCelsium(farengeitTypeDegree)
                    }}>°F</Button>
                </Dropdown>
              </div>
            </Col>
            <Col lg={{ span: 4, offset: 4 }} xs={12} md={6}>
              <div className="search-block">
                <InputGroup className="mb-3" value={this.state.inputValue} onChange={evt => this.updateInputValue(evt)}>
                  <FormControl
                    placeholder="Search"
                    type="search"
                    aria-label="Search"
                    aria-describedby="basic-addon2"
                  />
                  <InputGroup.Append>
                    <Button variant="outline-secondary" type='submit' onClick={() => { this.props.updateInputValue(this.state.inputValue)}}>Search</Button>
                  </InputGroup.Append>
                </InputGroup>
              </div>
            </Col>
          </Row>
        </Container>
      </div>
    </>
    );
  }
}

export default ControlBlock;
