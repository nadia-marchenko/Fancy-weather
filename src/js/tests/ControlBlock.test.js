/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import ControlBlock from '../components/ControlBlock';
/* eslint-enable no-unused-vars */

describe('>>>ControlBlock --- Snapshot', () => {
  it('+++capturing Snapshot of TodayWeather', () => {
    const renderedValue = renderer.create(<ControlBlock/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
