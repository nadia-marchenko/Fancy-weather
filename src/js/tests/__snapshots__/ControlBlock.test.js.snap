// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ControlBlock --- Snapshot +++capturing Snapshot of ControlBlock 1`] = `
<div
  className="control-block"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-md-6 col-12"
      >
        <div
          className="main-buttons"
        >
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Update
          </button>
           
          <div
            className="dropdown btn-group"
            onKeyDown={[Function]}
            role="group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-outline-secondary"
              disabled={false}
              id="dropdown-variants-Secondary"
              onClick={[Function]}
              type="button"
            >
              LAN
            </button>
          </div>
          <div
            className="temperature-change-buttons dropdown btn-group"
            onKeyDown={[Function]}
            role="group"
            variant="outline-secondary"
          >
            <button
              className="btn btn-outline-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              째C
            </button>
            <button
              className="btn btn-outline-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              째F
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-lg-4 col-md-6 col-12 offset-lg-4"
      >
        <div
          className="search-block"
        >
          <form
            className=""
          >
            <div
              className="mb-3 input-group"
              onChange={[Function]}
              value=""
            >
              <input
                aria-describedby="basic-addon2"
                aria-label="Search"
                className="form-control"
                placeholder="Search"
                type="search"
              />
              <button
                className="btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Search
              </button>
               
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`>>>ControlBlock --- Snapshot +++capturing Snapshot of TodayWeather 1`] = `
<div
  className="control-block"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-md-6 col-12"
      >
        <div
          className="main-buttons"
        >
          <button
            className="btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Update
          </button>
           
          <div
            className="dropdown btn-group"
            onKeyDown={[Function]}
            role="group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-outline-secondary"
              disabled={false}
              id="dropdown-variants-Secondary"
              onClick={[Function]}
              type="button"
            >
              LAN
            </button>
          </div>
          <div
            className="temperature-change-buttons dropdown btn-group"
            onKeyDown={[Function]}
            role="group"
            variant="outline-secondary"
          >
            <button
              className="btn btn-outline-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              째C
            </button>
            <button
              className="btn btn-outline-secondary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              째F
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-lg-4 col-md-6 col-12 offset-lg-4"
      >
        <div
          className="search-block"
        >
          <form
            className=""
          >
            <div
              className="mb-3 input-group"
              onChange={[Function]}
              value=""
            >
              <input
                aria-describedby="basic-addon2"
                aria-label="Search"
                className="form-control"
                placeholder="Search"
                type="search"
              />
              <button
                className="btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                Search
              </button>
               
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
